blocks/form/form.js

```diff
@@ -1,9 +1,49 @@
+const validityKeyMsgMap = {
+  typeMismatch: 'ErrorMessageInvalid',
+  badInput: 'ErrorMessageRequired',
+  patternMismatch: 'ErrorMessagePattern',
+  rangeOverflow: 'ErrorMessageMax',
+  rangeUnderflow: 'ErrorMessageMin',
+  tooLong: 'ErrorMessageMax',
+  tooShort: 'ErrorMessageMin',
+  valueMissing: 'ErrorMessageRequired',
+};
+
+function setElementProps(element, key, value) {
+  if (value) {
+    element.setAttribute(key, value);
+  }
+}
+
+function setConstraints(fd, element) {
+  if (fd.Mandatory === 'true') {
+    element.setAttribute('required', true);
+  }
+  setElementProps(element, 'pattern', fd.pattern);
```

Comment:
```
setElementProps(element, ...entries) {
  entries.filter((p, v) => v).forEach((p,v) => element.setAttribute(p, v))
}
```
and use it
```
setElementProps(element, Object.entries({
   'pattern' : fd.Pattern,
   'min' : fd.Min,
   'max' : fd.Max,
   'minlength' : fd.Min,
   'maxlength' : fd.Max
}))
```

Reply:
Min & Max only need to set on input type number & date for input type string, email, etc. need to set minlength & maxlength